examples/src/java/org/pantsbuild/example/README.md

```diff
@@ -208,13 +208,96 @@ contents a directory tree, a giant jar, or something else.
 Toolchain
 ---------
 
+### Ivy
+
 Pants uses [Ivy](http://ant.apache.org/ivy/) to resolve `jar` dependencies. To change how Pants
 resolves these, configure `resolve.ivy`.
 
+### Coursier
+
+Starting at release `1.4.0.dev26`, Pants added an option to pick [coursier](https://github.com/coursier/coursier)
+as the JVM 3rdparty resolver, with performance improvement being the main motivation. The goal is to retire ivy
+eventually.
+
+#### Limitation
+
+* Currently coursier only supports maven style repo resolves. [Resolving with ivy settings is still not mature](https://github.com/coursier/coursier/issues/created_by/benjyw).
+* Coursier does not do publishing.
+
+#### Example config to use coursier
+
+##### For Pants >= 1.7.x
+
+    :::ini
+    # This will turn on coursier and turn off ivy.
+    [resolver]
+    resolver: coursier
+
+    [resolve.coursier]
+    # jvm option in case of large resolves
+    jvm_options: ['-Xmx4g', '-XX:MaxMetaspaceSize=256m']
+
+    [export]
+    # Same if needed for large resolves
+    jvm_options: ['-Xmx4g', '-XX:MaxMetaspaceSize=256m']
+
+    [coursier]
+    repos: ['https://repo1.maven.org/maven2', 'https://dl.google.com/dl/android/maven2/']
+
+    # Change the following if you choose to [build coursier jar from scratch](https://github.com/coursier/coursier/blob/master/DEVELOPMENT.md#build-with-pants)
+    # or to fetch from different location.
+    bootstrap_jar_url: <url>
+    version: <version>
+
+* To inspect the resolve result, specify `--resolver-coursier-report`
+* To show coursier command line invocation, use `-ldebug`
+
+        :::bash
+        ./pants --cache-ignore --resolver-resolver=coursier resolve.coursier --report examples/tests/scala/org/pantsbuild/example/hello/welcome -ldebug
```

Comment:
Why is the `--cache-ignore` necessary here? Assuming this isn't to avoid a bug (as is the case with ivy), making `--report` and `--open` `fingerprint=True` options would be preferable.

Reply:
`--cache-ignore` means to ensure coursier is called, in case 
```
./pants --resolver-resolver=coursier resolve.coursier --report examples/tests/scala/org/pantsbuild/example/hello/welcome
```
is called twice in a row, then the 2nd run will just be validated. I wanted to use `invalidate` but that doesn't work anymore due to #5674

Reply:
>  then the 2nd run will just be validated

If the `--report` option is specified, then we should consider that a bug, IMO. Basically, the implementation of `--report` should be invoked outside of the invalidated block, because it is a sideeffect ... so I actually take back what I said about fingerprinting it: if it's outside the block, the fingerprint isn't relevant to it.

Reply:
Gotcha. will make `--report` essentially doing a forced run.